src/build_helper/src/git.rs

```diff
@@ -27,145 +28,196 @@ pub fn output_result(cmd: &mut Command) -> Result<String, String> {
     String::from_utf8(output.stdout).map_err(|err| format!("{err:?}"))
 }
 
-/// Finds the remote for rust-lang/rust.
-/// For example for these remotes it will return `upstream`.
-/// ```text
-/// origin  https://github.com/pietroalbani/rust.git (fetch)
-/// origin  https://github.com/pietroalbani/rust.git (push)
-/// upstream        https://github.com/rust-lang/rust (fetch)
-/// upstream        https://github.com/rust-lang/rust (push)
-/// ```
-pub fn get_rust_lang_rust_remote(
-    config: &GitConfig<'_>,
+/// Represents the result of checking whether a set of paths
+/// have been modified locally or not.
+#[derive(PartialEq, Debug, Clone)]
+pub enum PathFreshness {
+    /// Artifacts should be downloaded from this upstream commit,
+    /// there are no local modifications.
+    LastModifiedUpstream { upstream: String },
+    /// There are local modifications to a certain set of paths.
+    /// "Local" essentially means "not-upstream" here.
+    /// `upstream` is the latest upstream merge commit that made modifications to the
+    /// set of paths.
+    HasLocalModifications { upstream: String },
+    /// No upstream commit was found.
+    /// This should not happen in most reasonable circumstances, but one never knows.
+    MissingUpstream,
+}
+
+/// This function figures out if a set of paths was last modified upstream or
+/// if there are some local modifications made to them.
+/// It can be used to figure out if we should download artifacts from CI or rather
+/// build them locally.
+///
+/// The function assumes that at least a single upstream bors merge commit is in the
+/// local git history.
+///
+/// `target_paths` should be a non-empty slice of paths (relative to `git_dir` or the
```

Comment:
The "or" here is confusing -- how does it know relative to what they are?

Reply:
This function is always called with `Some`, so I will just make the directory required.

But to answer your question, the path from where git is executed was either `git_dir` (if set) or `cwd`. So the paths should be relative to either `git_dir` or `cwd`.

Reply:
Yeah the comment was inaccurate now, clarified that it is relative to `git_dir`. Thanks!